app/controllers/application_controller.rb

```diff
@@ -89,6 +90,14 @@ def set_gettext_locale
     super
   end
 
+  cattr_accessor :translations_refreshed_at
+  def possibly_reset_fastgettext_cache
+    ApplicationController.translations_refreshed_at ||= Time.now
```

Comment:
This is purely semantic, but I think we should store this as `refresh_translations_after` and then store the future time, that way, we can just check if `refresh_translations_after.blank? || Time.current > refresh_translations_after`.  It eases the cognitive overload.